tests/suites/storage/charm_storage.sh

```diff
@@ -0,0 +1,142 @@
+run_charm_storage() {
+	echo
+
+	model_name="test-storage"
+
+	echo "Assess create-storage-pool"
+	juju create-storage-pool -m "${model_name}" loopy loop size=1G
+	juju create-storage-pool -m "${model_name}" rooty rootfs size=1G
+	juju create-storage-pool -m "${model_name}" tempy tmpfs size=1G
+	juju create-storage-pool -m "${model_name}" ebsy ebs size=1G
+	echo "create-storage-pool PASSED"
+
+	# Assess the above created storage pools.
+	echo "Assessing storage pool"
+	if [ "${BOOTSTRAP_PROVIDER:-}" == "ec2" ]; then
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebs | .provider' | check "ebs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.["ebs-ssd"] | .provider' | check "ebs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tmpfs | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loop | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rootfs | .provider' | check "rootfs"
+	else
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rooty | .provider' | check "rootfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tempy | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loopy | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebsy | .provider' | check "ebs"
+	fi
+	echo "Storage pool PASSED"
+
+	# Assess charm storage with the filesystem storage provider
+	echo "Assessing filesystem rootfs"
+	juju deploy ./testcharms/charms/dummy-storage-fs --series="jammy" --storage data=rootfs,1G
+	wait_for "dummy-storage-fs" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage[] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[]' | check "data/0"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-fs/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
+	echo "Filesystem rootfs PASSED"
+	# remove the application
+	juju remove-application dummy-storage-fs
+	wait_for "{}" ".applications"
+
+	# Assess charm storage with the filesystem storage provider
+
```

Comment:
Do we need this empty line?

Reply:
We need to check for the `{}` for if we do not the next storage unit number will increment since the already previous storage unit have not been removed.

Reply:
EDIT: I will need to confirm this once more.

Reply:
I have confirmed, without the `wait_for` the test assertions fail why? Juju make block storages persistent and when you deploy with an existing persistent storage unit, then the next storage unit is incremented, hence the test assertion will fail. Yes I can use `remove-storage`, but that is not how the python test am replacing is currently implemented as it is does not test for that.

Reply:
I was talking about "empty line (number: 47)"  :)  not line with wait_for. 

To `wait_for` I've no doubt.

Reply:
Removed the newline.